### PR TITLE
Configuration: Pass #3

### PR DESCRIPTION
**Pull request by anurse**
As discussed https://github.com/aspnet/Docs/pull/7031#issuecomment-398875299 I made some changes and decided to just open a new PR to get things cleared up.

[Internal Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/configuration?view=aspnetcore-2.1&branch=pr-en-us-7199)

### CODE REVIEW

File: aspnetcore/signalr/configuration.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+In the JavaScript client, a similar `configureLogging` method exists. Provide a `LogLevel` value indicating the minimum level of log messages to produce. Logs will be written to the browser console window.
+
+```javascript
+let connection = new signalR.HubConnectionBuilder()
+    .withUrl("/myhub")
+    .configureLogging(signalR.LogLevel.Information)
+```
+
+> [!NOTE]
+> To disable logging entirely, specify `signalR.LogLevel.None` in the `configureLogging` method
```
Comment (rachelappel) [inline aspnetcore/signalr/configuration.md]: method.

File: aspnetcore/signalr/configuration.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+Client options can be configured on the `HubConnectionBuilder` type (available in both .NET and JavaScript clients), as well as on the `HubConnection` itself.
+
+### Configure logging
+
+Logging is configured in the .NET Client using the `ConfigureLogging` method. Logging providers and filters can be registered in the same way as they are on the server. See the [Logging in ASP.NET Core](xref:fundamentals/logging/index#how-to-add-providers) documentation for more information.
+
+> [!NOTE]
+> In order to register Logging providers, you must install the necessary packages. See the [Built-in logging providers](xref:fundamentals/logging/index#built-in-logging-providers) section of the docs for a full list.
+
+For example, to configure the console logger (after installing the `Microsoft.Extensions.Logging.Console` NuGet package):
```
Comment (rachelappel) [inline aspnetcore/signalr/configuration.md]: The for example sounds a bit out of place after a note and the paragraph before it.
Maybe
For example, to configure the console logger (after installing the `Microsoft.Extensions.Logging.Console` NuGet package):
->
The following example shows how to configure the console logger after installing the `Microsoft.Extensions.Logging.Console` NuGet package:

Comment (anurse) [reply aspnetcore/signalr/configuration.md]: How about:

For example, to enable Console logging, install the `Microsoft.Extensions.Logging.Console` NuGet package and call the `AddConsole` extension method:

File: aspnetcore/signalr/configuration.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+In the JavaScript client, transports are configured by setting the `transport` field on the options object provided to `withUrl`:
+
+```javascript
+let connection = new signalR.HubConnectionBuilder()
+    .withUrl("/myhub", { transport: signalR.HttpTransportType.WebSockets | signalR.HttpTransportType.LongPolling })
+    .build();
+```
+
+### Configure bearer authentication
```
Comment (davidfowl) [inline aspnetcore/signalr/configuration.md]: I think we need another article on this specifically with server side examples of consuming the token using the JWT auth handler

Comment (rachelappel) [reply aspnetcore/signalr/configuration.md]: I opened an issue for this.

File: aspnetcore/signalr/configuration.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+In the .NET Client, timeout values are specified as `TimeSpan` values. In the JavaScript client, timeout values are specified as numbers. The numbers represent time values in milliseconds.
+
+### Configure additional options
+
+Additional options can be configured in the `WithUrl` (`withUrl` in JavaScript) method on `HubConnectionBuilder`:
+
+| .NET Option | JavaScript Option | Description |
+| ----------- | ----------------- | ----------- |
+| `AccessTokenProvider` | `accessTokenFactory` | A function returning a string that is provided as a Bearer authentication token in HTTP requests. |
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: There are some options missing:
`ClientCertificates`
`CloseTimeout`
`HttpMessageHandlerFactory`
`Transports`
`Url`
`UseDefaultCredentials`

File: aspnetcore/signalr/configuration.md (hunk 3 above)
Comment (anurse) [reply aspnetcore/signalr/configuration.md]: We have #5925 tracking that

File: aspnetcore/signalr/configuration.md (hunk 4 above)
Comment (anurse) [reply aspnetcore/signalr/configuration.md]: For `Url` and `Transports`, I didn't want to confuse users by the fact that there are multiple ways to configure those options. The overloads allow them to be set just fine.

**Comment by anurse** [comment]
:up: :date: added missing options and inlined the samples (as per my comment earlier about none of these being part of a full sample)

**Comment by rachelappel** [comment]
LGTM

**Pull request merged**